CRM/Contact/Page/View/UserDashBoard.php

```diff
@@ -246,11 +243,12 @@ public static function &links() {
   /**
    * Get the user checksum from the url to use in links.
    *
-   * @return string
+   * @return string|false
+   * @throws \CRM_Core_Exception
    */
   protected function getUserChecksum() {
     $userChecksum = CRM_Utils_Request::retrieve('cs', 'String', $this);
-    if (empty($userID) && $this->_contactId) {
+    if ($this->_contactId && CRM_Contact_BAO_Contact_Utils::validChecksum($this->_contactId, $userChecksum)) {
```

Comment:
Related: https://lab.civicrm.org/dev/core/-/issues/2724 but I'd need to get my head back into it.

Reply:
Ah right - I assume that we could put back in a `getLoggedInUser` check there then

Reply:
@demeritcowboy I took a look at this again & I'm inclined to close https://lab.civicrm.org/dev/core/-/issues/2724 and keep this.

The usages of the return value are

1) to determine it the contact can add themselve to groups - if they used a checksum link in the url then no.
2) to decide whether to inclue a checksum in a url for the transact page - if checksum was used then yes. 

The impact of this PR is 
1 - unchanged. This does mean that if they authenticated by checksum AND are otherwise logged in they can't edit their groups. No-one noticed in the last year & my mind spins at what scenario this could matter in
2 - is slightly changed to 'if checksum was used **& it is valid** then yes by this pr